File: src/main/java/org/jdc/template/auth/MyAccountInterceptor.java
package org.jdc.template.auth;

import java.io.IOException;

import javax.inject.Inject;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class MyAccountInterceptor implements Interceptor {
    @Inject
    public MyAccountInterceptor() {
    }

    public Response intercept(Chain chain) throws IOException {
        Request.Builder builder = chain.request().newBuilder();

        // perform authentication

        return chain.proceed(builder.build());
    }
}

File: app/src/main/kotlin/org/jdc/template/auth/MyAccountInterceptor.kt
package org.jdc.template.auth

import java.io.IOException

import javax.inject.Inject

import okhttp3.Interceptor
import okhttp3.Response

class MyAccountInterceptor @Inject
constructor() : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val builder = chain.request().newBuilder()

        // perform authentication

        return chain.proceed(builder.build())
    }
}